tests/unit/test_parsers.py

```diff
@@ -1646,3 +1646,92 @@ def test_can_handle_generic_error_message(parser, body):
     )
     assert parsed['Error'] == {'Code': '503', 'Message': 'Service Unavailable'}
     assert parsed['ResponseMetadata']['HTTPStatusCode'] == 503
+
+
+@pytest.fixture
+def s3_parser():
+    return parsers.S3Parser()
+
+
+@pytest.fixture
+def expires_output_shape():
+    output_shape = model.StructureShape(
+        'OutputShape',
+        {
+            'type': 'structure',
+            'members': {
+                'Expires': {
+                    'shape': 'Expires',
+                    'location': 'header',
+                    'locationName': 'Expires',
+                }
+            },
+        },
+        model.ShapeResolver({'Expires': {'type': 'timestamp'}}),
+    )
+    return output_shape
+
+
+@pytest.mark.parametrize(
+    "expires, expected_expires",
+    [
+        (
+            'Thu, 01 Jan 2015 00:00:00 GMT',
+            datetime.datetime(2015, 1, 1, tzinfo=tzutc()),
+        ),
+    ],
+)
+def test_s3_valid_expires_response_parsed(
+    s3_parser, expires_output_shape, expires, expected_expires
+):
+    parser = s3_parser
+    output_shape = expires_output_shape
+    parsed = parser.parse(
+        {
+            'headers': {'Expires': expires},
+            'body': b'',
+            'status_code': 200,
+        },
+        output_shape,
+    )
+    expected = {
+        'ResponseMetadata': {
+            'HTTPStatusCode': 200,
+            'HTTPHeaders': {'expires': expires},
+        },
+        'Expires': datetime.datetime(2015, 1, 1, tzinfo=tzutc()),
+        'ExpiresString': expires,
+    }
+    assert parsed == expected
+
+
+@pytest.mark.parametrize(
+    "expires",
+    [
+        "Invalid Date",
+        "access plus 1 month",
+        "Expires: Thu, 9 Sep 2013 14:19:41 GMT",
+        "{ts '2023-10-10 09:27:14'}",
```

Comment:
Let's add another test case here for `-3370280040400`. This offset is going to trigger [the issue](https://github.com/boto/botocore/pull/3232/files#r1729384969) I pointed out above where it's falls within our Overflow error handling but doesn't trigger anything else. That falls through to the `RuntimeError` which will mean we can't create an `ExpiresString` entry.